src/configs.py

```diff
@@ -13,11 +14,25 @@
 class CharmConfig:
     """A class representing the data source of charm configurations."""
 
-    def __init__(self, config: ConfigData) -> None:
+    REQUIRED_KEYS = ["directory_api_url", "directory_api_token"]
+
+    def __init__(self, config: ConfigData, model: Model) -> None:
         self._config = config
+        self._model = model
+
+    def _get_directory_api_token(self) -> str:
+        secret_id = self._config["directory_api_token"]
+        secret = self._model.get_secret(id=secret_id)
+        return secret.get_content()[CONFIG_DIRECTORY_API_TOKEN_SECRET_KEY]
```

Comment:
question: would it be of any usefulness to add `refresh=True` to the `get_content()` method call to always get the latest revision of the secret?

Reply:
This makes sense. In general we are not refreshing secrets in our charms, but we should start adding this logic. 

AFAICT the ops framework will cache the secret unless `refresh=True` (see [here](https://ops.readthedocs.io/en/latest/reference/ops.html#ops.Secret.get_content)), but this behavior seems to be buggy and will be removed in the future (see [here](https://github.com/canonical/operator/issues/1536)).

Not sure if we should refresh the secret only when we get the `secret_changed` event or if it is fine to always set `refresh=True`, I will change it for now and try to ask for best practices.